src/main/webapp/app/exercises/shared/exercise/exercise.service.ts

```diff
@@ -146,15 +158,18 @@ export class ExerciseService {
      * Get exercise details including all results for the currently logged-in user
      * @param { number } exerciseId - Id of the exercise to get the repos from
      */
-    getExerciseDetails(exerciseId: number): Observable<EntityResponseType> {
-        return this.http.get<Exercise>(`${this.resourceUrl}/${exerciseId}/details`, { observe: 'response' }).pipe(
-            map((res: EntityResponseType) => {
-                this.processExerciseEntityResponse(res);
-
+    getExerciseDetails(exerciseId: number): Observable<EntityDetailsResponseType> {
+        return this.http.get<ExerciseDetailsType>(`${this.resourceUrl}/${exerciseId}/details`, { observe: 'response' }).pipe(
+            map((res: EntityDetailsResponseType) => {
                 if (res.body) {
+                    res.body.exercise = ExerciseService.convertExerciseDatesFromServer(res.body.exercise)!;
+                    ExerciseService.parseExerciseCategories(res.body.exercise);
                     // insert an empty list to avoid additional calls in case the list is empty on the server (because then it would be undefined in the client)
-                    if (res.body.posts === undefined) {
-                        res.body.posts = [];
+                    if (res.body.exercise.posts === undefined) {
+                        res.body.exercise.posts = [];
+                    }
```

Comment:
```suggestion
                    res.body.exercise.posts ??= [];
```

Reply:
Honestly, I had no idea that this was possible before and I also don't really think it's too intuitive. An optional assign...

I'd prefer to keep it this way if it's okay for you. If we use it throughout the codebase and I just didn't realise it until now, let me know and we can change this instance too.